core/src/Kokkos_Macros.hpp

```diff
@@ -198,9 +198,11 @@
 #endif
 #endif
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
 #if !defined(KOKKOS_MEMORY_ALIGNMENT)
 #define KOKKOS_MEMORY_ALIGNMENT 64
```

Comment:
We could try to emit warnings in case that macro is being used
```suggestion
#define KOKKOS_IMPL_MEMORY_ALIGNMENT 64
#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS 
#define KOKKOS_MEMORY_ALIGNMENT \
  [][[deprecated("KOKKOS_MEMORY_ALIGNMENT macro is deprecated")]]{ return KOKKOS_IMPL_MEMORY_ALIGNMENT; }()
#else
#define KOKKOS_MEMORY_ALIGNMENT KOKKOS_IMPL_MEMORY_ALIGNMENT
#endif
```

Reply:
That won't work with `gcc-8`, see https://godbolt.org/z/da1WhzjY4, but only from gcc-9.3 on but we can have a deprecated variable inside the lambda.